app/models/solr_document.rb

```diff
@@ -8,6 +8,16 @@ class SolrDocument
   # alias this field for the request link
   attribute :bibid, :string, "bibid_ssi"
 
+  def collection_identifier
+    level_ssm = self["level_ssm"]
+    unitid_ssm = self["unitid_ssm"]
+
+    level_ssm = level_ssm.join(" ") if level_ssm
```

Comment:
`join` is smart enough to not inject delimiters if an array is `nil`, empty or only has 1 value in it when you use `join`. `join` is also smart enough to not throw an exception if the array is `nil`. it also returns a new value instead of modifying the variable it's called on, so you don't need to assign this to a local variable.

Plus, the use of `compact` will exclude empty elements from the array anyway.

This whole method can be rewritten as:
```ruby
[level_ssm.join(" "), unitid_ssm.join(" ")].compact.join(", ")
```